src/hooks/useCustomLoaders.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import firebase from "firebase/app";
+
+import { CustomLoader, isCustomLoader } from "types/CustomLoader";
+
+import {
+  retrieveCustomLoaders as retrieveCachedCustomLoaders,
+  storeCustomLoaders as cacheCustomLoaders,
+} from "utils/localStorage";
+
+export interface CustomLoadersState {
+  customLoaders: CustomLoader[];
+  hasCustomLoaders: boolean;
+}
+
+export const initialValue: CustomLoadersState = {
+  customLoaders: [],
+  hasCustomLoaders: false,
+};
+
+export const CustomLoadersContext = createContext<CustomLoadersState>(
+  initialValue
+);
+
+export const CustomLoadersProvider: React.FC = ({ children }) => {
+  const [customLoaders, setCustomLoaders] = useState<CustomLoader[]>(
+    retrieveCachedCustomLoaders()
+  );
+
+  // Fetch the loaders data on first load
+  useEffect(() => {
+    (async () => {
+      const loadersSnapshot = await firebase
+        .firestore()
+        .collection("loaders")
+        .get();
+
+      const loaders: CustomLoader[] = loadersSnapshot.docs
+        .map((d) => d.data())
+        .filter(isCustomLoader);
```

Comment:
[TODO] Move this to `api/loaders` or similar?

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1240/commits/062f56ff18cfee034a4050f00b46535ed7160bad